### PR TITLE
Tsweb 1587 upd ai page

### PR DESCRIPTION
**Pull request by VeronikaDolbina**

### CODE REVIEW

File: ui_tests_playwright/automation/preconditionsData/technologyStack/TechnologyStackData.ts [hunk 1]
```diff
@@ -40,13 +40,15 @@ export default class TechnologyStackData {
...
+		'Data storage and\nmanipulation',
+		'DevOps',
+		'QA',
 		'Data Visualization',
-		'Data Storage',
-		'Development Environment',
+		'CI/CD and MLOps',
 		'Cloud',
-		'MLOps',
 	];
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/preconditionsData/technologyStack/TechnologyStackData.ts:52]: Most likely these changes will affect other pages, where this block is represented (eg Our Services page). Mb better leave this data as before and to provide custom titles only for AI page (if changes only on AI page )

File: ui_tests_playwright/automation/identifiers/mainSite/GeneralContainersMainSite.ts [hunk 2]
```diff
@@ -24,4 +24,7 @@ export default class GeneralContainersMainSite {
 	static CooperationModels = 'Container-CooperationModels';
 	static WhyTechstack = 'Container-WhyTechstack';
 	static WhatResultsCanYouExpect = 'Container-WhatResultsCanYouExpect';
+	static WhyPartnerWithUs = 'Container-WhyPartnerWithUs';
+	static CustomAiSolutions = 'Container-CustomAiSolutions';
+	static AiDevServToDeliverBusinessValue = 'Container-AiDevServToDeliverBusinessValue';
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/identifiers/mainSite/GeneralContainersMainSite.ts:29]: All these titles used on other pages, or they unique for AI page yet?

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 3]
```diff
@@ -34,37 +34,59 @@ test(
...
 		];
 
 		await baseDriverSteps.checkContainerTitlesAndNumbers(containers, expectedData);
 	}
 );
 
+test(
+	qase(
+		5604,
+		'Check block titles in "What Results Can You Expect with Our AI Expertise?" container from the "AI Development" page @desktop @mobile @Regression @AiDevelopment @TSWEB-694'
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:79]: mb better "Check section numbers ... "?

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 4]
```diff
@@ -124,6 +146,12 @@ test(
 		await expect(containerBlocks.nth(5).getByTestId(Container.BlockTitle)).toHaveText('Predictive analytics');
 		await expect(allSectionTitlesSixthBlock).toHaveText(testDataSixthBlock);
 
+		const allSectionTitlesSeventhBlock = containerBlocks.nth(6).getByTestId(Container.SectionTitle);
+		const testDataSeventhBlock = ['Overview', 'Implementation in the real world'];
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:150]: here you can use "testData" variable as data

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 5]
```diff
@@ -195,10 +223,29 @@ test(
 	}
 );
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:225]: Optional

could you pls transfer the 'Check section titles in "AI’s Beneficial Impact on Industries" container from the "AI Development" page @desktop @mobile @Regression @AiDevelopment @TSWEB-694' test here, in blocks order?

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 6]
```diff
@@ -240,6 +287,63 @@ test(
...
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:288]: Please update the 'Check section titles and CTA in "Our Approach" container from the "AI Development" page @desktop @mobile @Regression @AiDevelopment @TSWEB-694' test by adding checks for images, and update title also pls

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 7]
```diff
@@ -240,6 +287,63 @@ test(
...
+	async () => {
+		const aiDevServToDeliverBusinessValueContainer = driver.getByTestId(
+			AiDevelopment.AiDevServToDeliverBusinessValue
+		);
+		const allSectionTitles = aiDevServToDeliverBusinessValueContainer.getByTestId(Container.SectionTitle);
+		const testData = [
+			'Team augmentation',
+			'End-to-end product development',
+			'Tech advisory and consulting services',
+			'Scalability and flexibility',
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:340]: extra title

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 8]
```diff
@@ -240,6 +287,63 @@ test(
 	}
 );
 
+test(
+	qase(
+		5606,
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:292]: is test case id exist on Qase? You can either create appropriate test in Qase (pay attention to Id !) or just skip this and remove qase function from title **for NEW tests**

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts [hunk 9]
```diff
@@ -114,7 +114,7 @@ test(
 test(
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts:114]: Optional

Could you please locate the 'Check redirect by links in "AI’s Beneficial Impact on Industries" container from the "AI Development" page @desktop @mobile @Regression @AiDevelopment @TSWEB-694' test (and mb others) in appropriate order, according to page blocks, just for easier navigation

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts [hunk 10]
```diff
@@ -114,7 +114,7 @@ test(
...
 	qase(
 		4820,
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts:116]: pls update 'Check redirect by links in "Industries We Serve" container from the "AI Development" page @desktop @mobile @Regression @AiDevelopment @TSWEB-694' test title with new block name

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts [hunk 11]
```diff
@@ -114,7 +114,7 @@ test(
...
-		'Check carousel arrows click in "The Way We work" container from the "AI Development" page @desktop @Regression @AiDevelopment @TSWEB-694'
+		'Check carousel arrows click in "The Way We Work" container from the "AI Development" page @desktop @Regression @AiDevelopment @TSWEB-694'
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentActions.test.ts:117]: pls locate test in correct order (after Our approach block tests)

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/identifiers/mainSite/GeneralContainersMainSite.ts (hunk 2 above)
Comment (VeronikaDolbina) [reply ui_tests_playwright/automation/identifiers/mainSite/GeneralContainersMainSite.ts:29]: Unique. Transferred

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts (hunk 8 above)
Comment (VeronikaDolbina) [reply ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:292]: Yes, I added a test case

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts [hunk 12]
```diff
@@ -163,10 +185,26 @@ test(
...
+		const aiBeneficialImpactOnIndustriesContainer = driver.getByTestId(
+			AiDevelopment.AiBeneficialImpactOnIndustries
+		);
+		const allSectionTitles = aiBeneficialImpactOnIndustriesContainer.getByTestId(Container.SectionTitle);
+
+		const testData = ['69\n%', '73\n%', '82\n%', '$\n13\ntrillion'];
+		await expect(allSectionTitles).toHaveText(testData);
+	}
+);
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:203]: pls locate after the test for "Industries We Deliver AI Development Services To"

File: ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts (hunk 6 above)
Comment (Valerii-Kryvokhyzha) [reply ui_tests_playwright/automation/test/ui/mainSite/services/aiDevelopment/AiDevelopmentContent.test.ts:298]: @VeronikaDolbina pls add checks for images

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

**Review by Valerii-Kryvokhyzha — APPROVED** [review]

**Pull request merged**